ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouter.java

```diff
@@ -288,7 +291,11 @@ public void onCompletion(GetBlobResultInternal result, Exception exception) {
         } else if (result.getBlobResult != null) {
           logger.error("Unexpected result returned by background get operation to fetch chunk ids.");
         } else if (result.storeKeys != null) {
-          initiateBackgroundDeletes(result.storeKeys);
+          List<BackgroundDeleteRequest> deleteRequests = new ArrayList<>(result.storeKeys.size());
+          for (StoreKey storeKey : result.storeKeys) {
+            deleteRequests.add(new BackgroundDeleteRequest(storeKey, serviceId));
```

Comment:
The prefix is still added on in the BackgroundDeleteRequest constructor. So, if "blah" was provided as the serviceIdSuffix argument, the service ID used for the delete will be "ambry-background-delete-blah"